dandi/cli/cmd_organize.py

```diff
@@ -212,6 +214,11 @@ def _get_metadata(path):
 
     metadata = create_unique_filenames_from_metadata(metadata)
 
+    # create video file name and re write nwb file external files:
+    metadata = create_external_file_names(metadata)
```

Comment:
I think this "mode" should be optional (add `--rewrite` option with for now a single possible value `external-paths`) and be made "available" (otherwise - error) only in `files_mode='copy'` for now. In principle we could go another way and if such option is available we would document/assume that those files which need to be modified will be copied, and the rest (which do not need to be modified) -- acted upon according to `files_mode` setting.
WDYT @bendichter @satra @jwodder ?

Also relates to https://github.com/dandi/dandi-cli/issues/840 in that if it is not given, we should still do the check if there are external links (if such mode is not enabled) and error out in such a case.

Reply:
I don't understand why it would not work with `files_mode=move/symlink` also. In the current design the videos would be nested as follows:
```
sub-subjectid/
    sub-subjectid_ses-sessionid_ecephys.nwb
    sub-subjectid_ses-sessionid_ecephys/ # the videos sit beside the Dandi oganized nwb files
        uuid_external_file_0.mp4 # actual file or a symlink
```
Lets say we have created symlinks instead and have correctly renamed of nwbfiles' `external_file` arg. 
After that, would the Dandi upload operation not work with video symlinks and create the correct folder structure on the hosted dandiset?
I imagine if another user downloads that, the same folder would exist on the local drive and the `external_file` in the downloaded nwb files be valid again.

Reply:
I would assume that `external_file` field should be modified in the .nwb file itself, right? (then I guess ok for `move` but not for symlink or hardlink since would modify original file)

although I am confused how it is modified if file seems to be opened in `mode='r'` (read) mode:
https://github.com/dandi/dandi-cli/pull/841/files#diff-222d83ee68f525e4cf9e4b6b189b26401c17338b7b40c0d65b46e66968226c71R284

Reply:
yes, sorry I meant if the video files exist as symlinks/hardlink instead of copy/move. In case of the nwbfile it makes sense since we are modifying the original in case of symlink/hardlink.